fix: race condition when forwarding livechat by splitting subscription removal (#33381)
RocketChat/Rocket.Chat: .changeset/dry-taxis-cry.md

---
'@rocket.chat/meteor': patch
---

> Adjusts the findOneAndDelete method to insert into the trash collection before deletion to prevent a race condition that causes livechat users to get stuck in the agent's sidebar panel after being forwarded.

Reviewer: small wording suggestion to focus more on the issue being fixed other than the technical reason
```suggestion
Fixes a race condition that causes livechats to get stuck in the agent's sidebar panel after being forwarded.
```

Reviewer 2: Change `livechats` for `Livechat conversations` and we good
